ray-on-gke/tpu/kuberay-tpu-webhook/main.go

```diff
@@ -635,6 +607,41 @@ func init() {
 func main() {
 	flag.Parse()
 
+	// use in-cluster config if kubeConfig path is not passed as a flag
+	if KubeConfigPath == "" {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			panic(err)
+		}
+		client = kubernetes.NewForConfigOrDie(config)
+	} else {
+		config, err := clientcmd.BuildConfigFromFlags("", KubeConfigPath)
+		if err != nil {
+			panic(err)
+		}
+		client = kubernetes.NewForConfigOrDie(config)
+	}
+
+	// instantiate PodInformer for Ray worker pods in the GKE cluster
+	tweakListOptionsFunc := func(options *metav1.ListOptions) {
+		options.LabelSelector = "ray.io/node-type=worker"
+	}
+	factory := informers.NewFilteredSharedInformerFactory(client, 5*time.Minute, metav1.NamespaceAll, tweakListOptionsFunc)
```

Comment:
A default resync period of 5 minutes seems pretty long. Any specific reason for choosing 5 m?

Reply:
I chose an arbitrarily higher value to avoid constantly hitting the API server and slowing down the webhook, especially for larger clusters. My thought process was that Pods from workergroups are created/intercepted almost at the same time so a lower resync period shouldn't have too much effect. I can lower it down to a minute or two if that seems acceptable.